python/pyarrow/array.pyx

```diff
@@ -360,3 +425,35 @@ cdef class Table:
             data.append(<object> arr)
 
         return pd.DataFrame(dict(zip(names, data)), columns=names)
+
+    def name(self):
+        self._check_nullptr()
+        return frombytes(self.table.name())
+
+    def schema(self):
+        raise NotImplementedError()
+
+    def column(self, index_or_name):
```

Comment:
I would prefer only to pass indexes here -- retrieval by name can go in a different method. Can you also add a `__getitem__(self, i)` that invokes `self.column(i)`?